### PR TITLE
[Merged by Bors] - refactor: turn `posPart` into a notation class again

### PR DESCRIPTION
**Pull request by j-loreaux**
---

I would like to use this notation for the positive part of selfadjoint elements in a C⋆-algebra. The alternative would be to scope both notations, but the disadvantage of that is that sometimes I would like to work with the notation both for the elements of the C⋆-algebra, and for real-valued functions in the same proof.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by YaelDillies** [comment]
That's not a `feat`!

Could you please move this notation to `Algebra.ZeroOne` (which you could rename `Algebra.Notation)? I have the plan of making `abs` notation again so that we can prove properties of `abs` on `Int` without importing ordered algebra.

File: Mathlib/Algebra/Order/Notation.lean [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+* `a⁺ = a ⊔ 0`: *Positive component* of an element `a` of a lattice ordered group
+* `a⁻ = (-a) ⊔ 0`: *Negative component* of an element `a` of a lattice ordered group
+-/
+
+/-- A notation class for the *positve part* function: `a⁺`. -/
+class PosPart (α : Type*) where
+  /-- The *positive part* of an element `a`. -/
+  posPart : α → α
+
+/-- A notation class for the *positve part* function (multiplicative version): `a⁺ᵐ`. -/
```
Comment (YaelDillies) [inline Mathlib/Algebra/Order/Notation.lean:24]:
```suggestion
/-- A notation class for the *positive part* function (multiplicative version): `a⁺ᵐ`. -/
```
Same above

File: Mathlib/Algebra/Order/Notation.lean [hunk 2]
```diff
@@ -0,0 +1,49 @@
+/-
+Copyright (c) 2024 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+
+import Mathlib.Tactic.ToAdditive
```
Comment (YaelDillies) [inline Mathlib/Algebra/Order/Notation.lean:7]:
```suggestion
import Mathlib.Tactic.ToAdditive

```

**Review by YaelDillies — APPROVED** [review]
maintainer merge

**Comment by kim-em** [comment]
bors merge

**Pull request closed without merging**